### PR TITLE
Add a combinator that transforms one array to another

### PR DESCRIPTION
**Pull request by vecerek**
Fully addresses https://github.com/Effect-TS/schema/issues/352 in combination with https://github.com/Effect-TS/schema/pull/359 as opposed to the approach outlined in https://github.com/Effect-TS/schema/pull/353.

### CODE REVIEW

File: src/Schema.ts [hunk 1]
```diff
@@ -2354,6 +2354,23 @@ export const itemsCount = <A>(
...
 
+/**
+ * This combinator transforms an `array` of type A into an `array` of type B given a schema that transforms values from A to B.
+ *
+ * @param self - The schema representing the input array
+ * @param of - The schema representing the transformation from A to B
+ * @category array
+ * @since 1.0.0
+ */
+export const arrayOf = <A, B>(of: Schema<A, B>) =>
```
Comment (bigpopakap) [inline src/Schema.ts:2365]: Love it! Again, maybe this should be a dual API?

Also, good choice of function name... which genius came up with that? 😉 😂

**Review by bigpopakap — COMMENTED** [review]
I'll have to leave it to the maintainers for final approval, but this all looks great to me. Thanks for contributing! I'm sure I'll benefit from these in the future :)

File: src/Schema.ts (hunk 1 above)
Comment (vecerek) [reply src/Schema.ts:2365]: Yeah, I liked your suggestion at least 10x more than mine 😅

**Comment by vecerek** [comment]
Closed by https://github.com/Effect-TS/schema/pull/361

**Pull request closed without merging**